stan/math/prim/mat/fun/gp_exp_quad_cov.hpp

```diff
@@ -34,28 +38,28 @@ template <typename T_x, typename T_sigma, typename T_l>
 inline
     typename Eigen::Matrix<typename stan::return_type<T_x, T_sigma, T_l>::type,
                            Eigen::Dynamic, Eigen::Dynamic>
-    gp_exp_quad_cov(const std::vector<T_x>& x, const T_sigma& sigma,
-                    const T_l& length_scale) {
+    gp_exp_quad_cov(const std::vector<T_x> &x, const T_sigma &sigma,
+                    const T_l &length_scale) {
   using std::exp;
-  check_positive("gp_exp_quad_cov", "marginal variance", sigma);
-  check_positive("gp_exp_quad_cov", "length-scale", length_scale);
+  check_positive("gp_exp_quad_cov", "magnitude", sigma);
+  check_positive("gp_exp_quad_cov", "length scale", length_scale);
   for (size_t n = 0; n < x.size(); ++n)
     check_not_nan("gp_exp_quad_cov", "x", x[n]);
 
   Eigen::Matrix<typename stan::return_type<T_x, T_sigma, T_l>::type,
                 Eigen::Dynamic, Eigen::Dynamic>
       cov(x.size(), x.size());
 
-  int x_size = x.size();
+  size_t x_size = x.size();
   if (x_size == 0)
     return cov;
 
   T_sigma sigma_sq = square(sigma);
   T_l neg_half_inv_l_sq = -0.5 / square(length_scale);
 
-  for (int j = 0; j < (x_size - 1); ++j) {
+  for (size_t j = 0; j < (x_size - 1); ++j) {
```

Comment:
Can you remove the parens? I don't think it makes it any clearer.
`j < x_size - 1`

Reply:
Can you also make this loop just `size_t j = 0; j < x_size; ++j` and remove the explicit setting of the last diagonal element outside the loop? I think most programmers would expect the loop to set the last diagonal.